TombEngine/Objects/TR3/Entity/WaspMutant.cpp

```diff
@@ -0,0 +1,289 @@
+#include "framework.h"
+#include "Objects/TR3/Entity/WaspMutant.h"
+
+#include "Game/control/box.h"
+#include "Game/effects/effects.h"
+#include "Game/effects/spark.h"
+#include "Game/misc.h"
+#include "Math/Math.h"
+#include "Specific/setup.h"
+
+using namespace TEN::Effects::Spark;
+using namespace TEN::Math;
+
+namespace TEN::Entities::Creatures::TR3
+{
+	constexpr auto WASP_DAMAGE = 50;
+
+	constexpr auto WASP_DETECTION_RANGE = SQUARE(BLOCK(15));
+	constexpr auto WASP_ATTACK_RANGE	= SQUARE(BLOCK(0.5f));
+
+	constexpr auto WASP_LAND_CHANCE = 1 / 256.0f;
+
+	constexpr auto WASP_LAND_TURN_RATE_MAX = ANGLE(1.0f);
+	constexpr auto WASP_AIR_TURN_RATE_MAX  = ANGLE(3.0f);
+
+	constexpr auto WASP_IDLE_TO_FLY_VELOCITY = BLOCK(1 / 20.0f);
+	constexpr auto WASP_VENOM_SACK_LIGHT_POWER = 10;
+
+	constexpr auto WaspVenomSackLightColor = Vector4(0.0f, 0.35f, 0.0f, 1.0f);
+
+	const auto WaspBite			 = BiteInfo(Vector3(0.0f, 0.0f, -260.0f), 12);
+	const auto WaspVenomSackBite = BiteInfo(Vector3::Zero, 10);
+
+	enum WaspMutantState
+	{
+		WASP_STATE_FLY_IDLE,
+		WASP_STATE_FLY_IDLE_TO_IDLE, // Floor
+		WASP_STATE_IDLE,			 // Floor
+		WASP_STATE_IDLE_TO_FLY_IDLE,
+		WASP_STATE_ATTACK,			 // Flying
+		WASP_STATE_FALL,
+		WASP_STATE_DEATH,
+		WASP_STATE_FLY_FORWARD
+	};
+
+	enum WaspMutantAnim
+	{
+		WASP_ANIM_FLY_IDLE,
+		WASP_ANIM_FLY_IDLE_TO_IDLE,
+		WASP_ANIM_IDLE,
+		WASP_ANIM_IDLE_TO_FLY_IDLE,
+		WASP_ANIM_ATTACK,
+		WASP_ANIM_FALL,
+		WASP_ANIM_DEATH,
+		WASP_ANIM_FLY_FORWARD
+	};
+
```

Comment:
Maybe the FLY_IDLE could be just IDLE 
and the current IDLE could be LANDING?

Because as I see, fly_idle is the base of the state machine but the ground idle is more like a secondary state of this creature.